.travis.yml

```diff
@@ -158,16 +158,7 @@ before_script: |
   fi
 
 script:
-  # each script we want to run need to go in it's own section and the program you want
-  # to fail travis need to be the last thing called
-  - |
-    echo "Calling pytest with the following arguments: $PYTEST_ADDOPTS"
-    python -mpytest
-  - |
-    if [[ $RUN_FLAKE8 == 1 ]]; then
-      flake8 --statistics && echo "Flake8 passed without any issues!"
-    fi
-
```

Comment:
Or just (approximately)
```
- |
  ([[ $RUN_FLAKE8 != 1 ]] || flake8 ... && echo "flake8 passed") &&
    echo "Called with $PYTEST_ADDOPTS" &&
    python -mpytest
```
?
Seems simpler than relying on travis' API...

Reply:
I'm just relying on the shell. `.travis-script.sh` is just a bash script no travis API involved.

I find it simpler to be able to use a multi-line script instead of trying to jam all the logic in a single expression.

Reply:
I personally prefer not splitting the CI scripts over multiple files.
Instead of putting everything into a single expression you can also use `set -e` or add `|| exit 1` after every (relevant) line.

Reply:
Also, if you *really* want to move it to its own script, I guess it should go to ci/travis-script.sh.